integration/container/create_test.go

```diff
@@ -0,0 +1,140 @@
+package container
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/integration-cli/environment"
+	"github.com/docker/docker/integration-cli/fixtures/load"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	testEnv *environment.Execution
+)
+
+func TestMain(m *testing.M) {
+	var err error
+	testEnv, err = environment.New()
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	if testEnv.LocalDaemon() {
+		fmt.Println("INFO: Testing against a local daemon")
+	} else {
+		fmt.Println("INFO: Testing against a remote daemon")
+	}
+
+	// TODO: ensure and protect images
+	// 2. Run the test(s)
+	res := m.Run()
+	os.Exit(res)
+}
+
+func TestAPICreateWithNotExistImage(t *testing.T) {
+	defer setupTest(t)()
+	clt := createClient(t)
+
+	testCases := []struct {
+		image         string
+		expectedError string
+	}{
+		{
+			image:         "test456:v1",
+			expectedError: "No such image: test456:v1",
+		},
+		{
+			image:         "test456",
+			expectedError: "No such image: test456",
+		},
+		{
+			image:         "sha256:0cb40641836c461bc97c793971d84d758371ed682042457523e4ae701efeaaaa",
+			expectedError: "No such image: sha256:0cb40641836c461bc97c793971d84d758371ed682042457523e4ae701efeaaaa",
+		},
+	}
+
+	for index, tc := range testCases {
+		tc := tc
+		t.Run(fmt.Sprintf("%d", index), func(t *testing.T) {
+			t.Parallel()
+			_, err := clt.ContainerCreate(context.TODO(),
```

Comment:
Do you expect to plumb context through the test cases at some point? Otherwise, I think `TODO` is not the right context to use here (we should probably have `ctx := context.Background()` above.